chromium_src/third_party/blink/renderer/core/dom/document.cc

```diff
@@ -28,7 +28,10 @@ namespace {
 //  frame that the given document is in. This includes frames that
 //  are disconnect, remote or local to the top level frame.
 std::string TopETLDPlusOneForDoc(const Document& doc) {
-  const auto host = doc.TopFrameOrigin()->Host();
+  const auto origin = doc.TopFrameOrigin();
+  if (!origin)
+    return "";
```

Comment:
I think this can be unsafe in certain cases and you should use `SecurityOrigin::UrlWithUniqueOpaqueOrigin` cc @diracdeltas

Reply:
technically this should ultimately return the string "null" since that's the serialization of an opaque origin. https://html.spec.whatwg.org/multipage/origin.html#ascii-serialisation-of-an-origin